### ISSUE TITLE
Screen reader issue with floating labels

### ISSUE THREAD

**Issue by HalHaynes** [twbs/bootstrap#41362]
### Prerequisites

- [x] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [x] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

The floating labels are lovely but there's an accessibility issue for screen readers (at least for NVDA on Chrome in Windows). Because the label is required to come after the input field, the screen reader reads it after it reads the field's contents. The label should be read first. When I position the `label` tag above the `input` tag, it breaks  the display of the control.

### Reduced test cases

I have no way to show a fix, I'm just reporting a problem with the code when written exactly as prescribed in the instructions for floating labels.

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.3.3

**Comment by julien-deramond** [comment]
Thanks for reporting this issue @HalHaynes, we'll look into the accessibility concern you've raised.
You're absolutely right about the `e.g.` versus `i.e.` usage. That correction is being addressed in https://github.com/twbs/bootstrap/pull/41364 and will be included in the next release.

**Comment by RichardD2** [comment]
Seems like [the `:has()` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) would be a good alternative for recent *(December 2023 or later)* browsers. Older brosers would probably need a JS-fallback.

For example, instead of:

```
.form-floating > .form-control:focus ~ label,
.form-floating > .form-control:not(:placeholder-shown) ~ label,
.form-floating > .form-control-plaintext ~ label,
.form-floating > .form-select ~ label
```

you could have:

```
.form-floating:has(> .form-control:focus) > label,
.form-floating:has(> .form-control:not(:placeholder-shown)) > label,
.form-floating:has(> .form-control-plaintext) > label,
.form-floating:has(> .form-select) > label
```

Then you could place the `<label>` before the `<input>`, and it would still work as expected.

**Comment by julien-deramond** [comment]
While I haven’t checked the use of `:has` in this precise context yet, this is likely one of the use cases where `:has` could be useful. It's currently not supported due to our existing `.browserslistrc` configuration. However, we're planning to revisit that for v6, so `:has` will probably be usable then.
